### ISSUE TITLE
Add support for interrupted map projections

### ISSUE THREAD

**Issue by PaulWessel** [GenericMappingTools/gmt#6137]
**Description of the desired feature**

This issues started with a [GMT Julia discussion](https://discourse.julialang.org/t/an-internal-julia-plotting-package/73079/47?u=joa-quim).  It is a lot of work to implement an interrupted projection in GMT since the map boundaries are all over and lines and polygons should be properly clipped across these many boundaries.  The alternative is to have a single module that just handles interrupted projections (piecemeal) and accepts the gmt command(s) you wish to replicate across the pieces.  I have implemented this (more or less) as a bash script, and can easily make maps like these.  Here is the land-centric Goode interrupted homolosine projection of our 2m DEM with shading:

![homolosine_land](https://user-images.githubusercontent.com/26473567/146867970-96c7ad20-0f1c-41aa-bf9f-5ef7a17df8ab.jpg)

Similarly, here are crustal ages on the ocean-centric Goode interrupted homolosine projection:

![homolosine_ocean](https://user-images.githubusercontent.com/26473567/146867984-9c05b3b1-875b-47df-83f7-d1e26c11f2af.jpg)

I imagine a syntax like this:

**gmt interrupt -T**_projtype/scale_ _-F_"cmd1; cmd2; ..." [ **-O -K -P -X -Y**]

where one can stack more than one command to be replicated at the various pieces involved.  Above for the DEM, the cmd was just "grdimage @earth_relief_02m -I+d". Given the unlimited number of cmd one can stack, one can plot grids or images and overlay land or data points, lines, or polygons or text or vectors etc.

Implementing this will be _simple_; the alternative is _hard_.  Pinging @GenericMappingTools/core for feedback. For instance, not clear to me how the **-F** option would be wrapped in the externals.

**Comment by joa-quim** [comment]
Hmm, why can't we have that **interrupt** module work silently under the hood and on surface we would still use ```grdimage -J<good_homolosine_land[_lon_0]>/scale ...```
and the same for other modules?

**Comment by PaulWessel** [comment]
Yes, perhaps.  Each plotting module would need to have a check if interrupted projection, then set up a call to **gmtinterrupt** and exit.  That call would itself make 3-12 calls to the same plot module with different -R -J settings per slice.  I will explore that.  This would mean no **interrupt** module per se, just a function that behaves like one.

**Comment by PaulWessel** [comment]
Accepting suggestions for how to specify these projections.  I think you said `igh_o` was the ocean one and there is no `igh_l` for land?  Is there an `igh` for non-interrupted homolosine where one can specify the central longitude? What about interrupted sinusoidal (3 pieces like in the old GMT logo)?  If there are official proj4 codes for these then we should use them.  If not then we need to make something up.  While I may not have the time or energy to try to implement dymaxion and Spielhaus projection, we should expect these codes as well eventually.

**Comment by joa-quim** [comment]
[See](https://proj.org/operations/projections/index.html)
After all they have the land version too. It's the ``igh``. The non-interrupted is ``good``
There's nothing in that page on other Interrupted projections.

**Comment by PaulWessel** [comment]
Thanks, I like the coastlines that wraps cross the map for the ocean case (!).  So no dymexian or interrupted sinusoidal means we need syntax for those.

**Comment by anbj** [comment]
Could you explain what a 'interrupted map projection' is?

**Comment by Esteban82** [comment]
Maybe what [wikipedia](https://en.wikipedia.org/wiki/Interruption_(map_projection)) says?

**Comment by anbj** [comment]
Thanks Esteban. How the heck did I not manage to google this before I asked.

**Comment by zwillikon** [comment]
I'm sorry to hear that Paul has deceased. Is there some chance for this feature left?

**Comment by joa-quim** [comment]
I think its chances lie on the general case of having more projections in GMT, which we already can do when using the proj4 syntax in -J. The problem is that we don't have code to create the frames & annotations for those external projections.
